generators/server/prompts.js

```diff
@@ -309,7 +309,9 @@ function askForServerSideOpts() {
         this.packageName = this.jhipsterConfig.packageName = props.packageName;
         this.serverPort = this.jhipsterConfig.serverPort = props.serverPort || '8080';
         this.cacheProvider = this.jhipsterConfig.cacheProvider = !reactive ? props.cacheProvider : 'no';
-        this.enableHibernateCache = this.jhipsterConfig.enableHibernateCache = !['no', 'memcached'].includes(this.cacheProvider) ? props.enableHibernateCache : false;
+        this.enableHibernateCache = this.jhipsterConfig.enableHibernateCache = !['no', 'memcached'].includes(this.cacheProvider)
+            ? props.enableHibernateCache
+            : false;
```

Comment:
@MarlonLuan can you try this instead?
The condition is not complete, so let's force false instead of undefined.
```suggestion
        this.enableHibernateCache = this.jhipsterConfig.enableHibernateCache = !!props.enableHibernateCache;
```

Reply:
Replace all the current code? And in relation to the condition
```javascript
cacheProvider == 'no'
```
, where it should be considered?

Reply:
When cacheProvider is `no`, enableHibernateCache should be `undefined`, double negate should convert `undefined` to `false`.

Reply:
I didn't understand the need for the change, according to the current code,:

when "cacheProvider == no" --> "enableHibernateCache == false", 

when "cacheProvider !== no" --> "enableHibernateCache ==  value of props.enableHibernateCache (informed by the prompt)", 

props.enableHibernateCache will not be undefined

Reply:
The condition to skip enableHibernateCache question is more complex then your condition.
https://github.com/jhipster/generator-jhipster/blob/8a1f652d2e1ad127631a8dc23a42cbc4ae460d07/generators/server/prompts.js#L278-L281.

In some cases, the answered value will be overridden by the condition.